python/src/nanoarrow/_array.pyx

```diff
@@ -189,13 +196,48 @@ cdef class CArrayView:
 
     @property
     def n_buffers(self):
+        if _types.is_data_view(self._ptr.storage_type):
+            return 2 + self._ptr.n_variadic_buffers + 1
+
         return self.layout.n_buffers
 
-    def buffer_type(self, int64_t i):
+    def _buffer_info(self, int64_t i):
         if i < 0 or i >= self.n_buffers:
             raise IndexError(f"{i} out of range [0, {self.n_buffers}]")
 
-        buffer_type = self._ptr.layout.buffer_type[i]
+        if (
+            _types.is_data_view(self._ptr.storage_type)
+            and i == (2 + self._ptr.n_variadic_buffers)
+        ):
+            return (
+                NANOARROW_BUFFER_TYPE_DATA,
```

Comment:
Checking the type of the variadic buffer is actually missing from the C++ test; we might want to go back and add this there too